### ISSUE TITLE
My Zcash wallet does not retain opids from z_sends (selective disclosure)

### ISSUE THREAD

**Issue by SterL** [zcash/zcash#1937]
It recently occurred to me that opids are not retained by my Zcash wallet when I perform the z_sendmany operation. From a business perspective, it would be super convenient if the wallet would store these opids (and the data about their correlated tx) because they may be critical in locating a customers funds. Otherwise it becomes the responsibility of myself (or someone in a similar situation) to store these opids correlated tx data separately from the wallet, and that is not entirely obvious at first, coming from a bitcoin background. Opids (and their tx data) need to be stored so that when a z_sendmany fails, we can query zcashd and determine the status of that particular tx at a much later time.

Here is an example to (hopefully) better illustrate the scenario that I'm talking about: 
1.) A customer has interacted with my system such that they are now waiting on the system to send them ZEC to a zaddr.
2.) The system sends the ZEC out with a z_sendmany and gets back the opid
3.) The system uses the opid to query the Zcash daemon until the the transaction is completed.
4.) Normally, the system will return the txid to the user and that's it. This is the case however, when a transaction fails and a txid is never returned.
5.) Customer contacts me in the near/far future to find out where their Zcash is.
6.) To verify the users request, I lookup their order info and use the opid stored in the order info to get information about said 'failed' tx from the zcashd. If there is a txid in that opid data, then i know the user is lying and trying to game me. If there is not a txid in the opid data, I know the user is telling the truth and i can manually send them their ZEC or give them a return or whatever. This also gives me the chance to explain to the customer in their support ticket why this transaction failed.

TL;DR If zcashd does not store the opid data, I can't query it in the future to check on a transactions status

**Comment by tromer** [comment]
Duplicate of #1929?

**Comment by bitcartel** [comment]
@sterl Thanks. Btw, typo in step 2 (I think you meant z_sendmany instead of z_sendtoaddress).

**Comment by SterL** [comment]
Thank you for the correction @bitcartel ! I edited my original post to reflect it.

**Comment by bitcartel** [comment]
Logging extended details about z related transactions could also apply to other rpc calls, such as z_shield proposed in #1785

**Comment by nathan-at-least** [comment]
I added #2057 as a specific way to address this concern. @SterL, let us know if #2057 is sufficient.

**Comment by SterL** [comment]
Yep. I think this should be sufficient for what we need. Just some way to go back and see an opid, if we may have missed it. Being able to grep by either an amount or an address is perfect 👍

**Comment by str4d** [comment]
@SterL has your use-case been addressed by us merging #2058?

**Comment by SterL** [comment]
@str4d yes it has! Thank you for checking in. I can now grep the debug.log using a zaddr, and it returns the associated opid's that I'm looking for 😊  Thanks again Zcash team!
